src/payment/strategies/modal/modal-payment-strategy.ts

```diff
@@ -0,0 +1,87 @@
+import { CheckoutStore, InternalCheckoutSelectors } from '../../../checkout';
+import { InvalidArgumentError } from '../../../common/error/errors';
+import { OrderActionCreator, OrderRequestBody } from '../../../order';
+import { OrderFinalizationNotRequiredError } from '../../../order/errors';
+import { PaymentArgumentInvalidError } from '../../errors';
+import PaymentActionCreator from '../../payment-action-creator';
+import { PaymentInitializeOptions, PaymentRequestOptions } from '../../payment-request-options';
+import PaymentStrategy from '../payment-strategy';
+
+import ModalPaymentInitializeOptions from './modal-payment-initialize-options';
+
+export default class ModalPaymentStrategy implements PaymentStrategy {
```

Comment:
I understand what you're trying to do. But to me, it's an abstraction for a very specific way of capturing payment information inside a modal. It makes several assumptions on how the iframe is provided. For example, it assumes the creation of the iframe is handled by the caller and the submission happens inside the iframe. For other payment methods, it could be handled by their own JS client or the submission happens outside of the iframe (i.e.: Braintree or Adyen). So for the latter, this strategy cannot be used. So it's not right to give it a very generic name `ModalPaymentStrategy` when it is not generic enough to handle different kinds of modal payment flows. I suggest you either give it a more specific name so it's clear to other developers that it's only usable in certain scenarios.